### ISSUE TITLE
Introduction to Web Development: <Link in the assignment to "Why Learning to Code is So Damn Hard" is a broken link, the page cannot be found.>

### ISSUE THREAD

**Issue by K4rr4s** [TheOdinProject/curriculum#28018]
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### Describe your suggestion

In the 1st Assignment on "https://www.theodinproject.com/lessons/foundations-introduction-to-web-development" point 1 has a link that is supposed to go to a web.archive.org page for "Why Learning Code is So Damn Hard", this doesn't appear to exist anymore so the link is broken. (The original page doesn't exist either.) URL : "https://web.archive.org/web/20230630111131/https://www.thinkful.com/blog/why-learning-to-code-is-so-damn-hard/"

### Path

Foundations

### Lesson Url

https://www.theodinproject.com/lessons/foundations-introduction-to-web-development

### (Optional) Additional Comments

This is on the 1st question on the 1st Assignment for the Foundations.

**Comment by ssimkani** [comment]
The first time I clicked on the link I got a 404 error, but now it works fine for me on my Mac.

**Comment by sukairaida** [comment]
Doesn't work for me either right now and when I was on that lesson I had to reload that page many times for it to finally load. Maybe it could be replaced with this [medium article](https://medium.com/@andrewlatta/why-learning-to-code-is-so-damn-hard-303eae632820) that seems to have the same content and works fine.

**Comment by K4rr4s** [comment]
Thanks for looking in to this! 

Turns out DNS was the answer... 
"web.archive.org" is blocked on my pi-hole, which uses the standard firebog ad-lists and pages on web.archive.org are blocked for security reasons. 
As I understand it, there is an exploit that gets round blocked Domain addresses by archiving them on web.archive.org. 
If I safelist the domain (or temporarily disable blocking) the page does load, though it did give a "404" pop-over error after it had been loaded for a little while, which looked like it was related to a missing Advert (ironically).

The Medium article sukairaida has suggested appears to do the job nicely and is a little more secure.

**Comment by ssimkani** [comment]
I'd be happy to submit a pull request updating the link!

**Comment by JoshDevHub** [comment]
The main thing I don't really like about the medium article is that it's not written by the same guy. It's a copy/paste job with no credit given to the original author (Erik Trautman, who is the creator of The Odin Project).

I'll ask the team what they think, but I'll also say that there are several other webarchive links throughout the curriculum (a quick search tells me nearly 40 links), so even if we fix this one, you'd eventually have to whitelist webarchive anyway.

**Comment by MidlifeMechanicus** [comment]
I am getting a 503 error trying to access the original link, when I have not had problems with the other archived pages that I have tried to access. Given the author of the piece, is it worth dedicating a page to hosting it here on TOP, so credit can be properly given?
